### PR TITLE
[freeathome] Fix not updating values of room temperature devices

### PR DESCRIPTION
**Pull request by JankKeks**
# Description
Fix for problems between freeathome and openhab. With 4.2. (and newer) things values was no longer sucessfull updated. Problem was caused by a bug generating the ID of the device Channels. (That leads to UpdateStatus failed).
[Bug](https://github.com/openhab/openhab-addons/issues/17953) should be fixed

Further Discussions (and remaining bugs..:
https://community.openhab.org/t/abb-busch-jager-free-home-official-rest-api/141698/119

### CODE REVIEW

File: bundles/org.openhab.binding.freeathome/src/main/java/org/openhab/binding/freeathome/internal/handler/FreeAtHomeDeviceHandler.java [hunk 1]
```diff
@@ -589,6 +587,12 @@ private void reloadChannelTypes() throws FreeAtHomeGeneralException {
         }
     }
 
+    // Create a channel UID. Makes sure that the channel UID is unique and generated the same way every time
+    private ChannelUID createChannelUID(ThingUID thingUID, String channelID, String dpgLabel) {
+        ChannelUID channelUID = new ChannelUID(thingUID, channelID, dpgLabel.substring(4));
+        return channelUID;
```
Comment (lsiepel) [inline bundles/org.openhab.binding.freeathome/src/main/java/org/openhab/binding/freeathome/internal/handler/FreeAtHomeDeviceHandler.java:593]:
```suggestion
        return new ChannelUID(thingUID, channelID, dpgLabel.substring(4));
```

**Review by lsiepel — CHANGES_REQUESTED** [review]
Thanks for fixing this issue! Hopefully this is a start for some more fixes.
Besides the minor style comment, i wonder why the binding folder was renamed, could you revert that renaming?

**Comment by jlaur** [comment]
> Besides the minor style comment, i wonder why the binding folder was renamed, could you revert that renaming?

I think it needs to be rebased so this will not be shown as differences. The actual binding id is `freeathome`, it was already renamed from `freeathomesystem`.

**Comment by lsiepel** [comment]
> > Besides the minor style comment, i wonder why the binding folder was renamed, could you revert that renaming?
> 
> I think it needs to be rebased so this will not be shown as differences. The actual binding id is `freeathome`, it was already renamed from `freeathomesystem`.

haha i thought it was the other way around... and guess what,.... i did it myself: https://github.com/openhab/openhab-addons/pull/16718

**Comment by JankKeks** [comment]
> > > Besides the minor style comment, i wonder why the binding folder was renamed, could you revert that renaming?
> > 
> > I think it needs to be rebased so this will not be shown as differences. The actual binding id is `freeathome`, it was already renamed from `freeathomesystem`.
> 
> haha i thought it was the other way around... and guess what,.... i did it myself: https://github.com/openhab/openhab-addons/pull/16718

Thanks for review. You renamed the binding already, but the internal path was not renamed - that caused a lot of internal warnings and problems with the mapping and classes. So I fixed it also.

**Comment by jlaur** [comment]
> You renamed the binding already, but the internal path was not renamed - that caused a lot of internal warnings and problems with the mapping and classes. So I fixed it also.

Thanks for clarifying and fixing this. Would you be able to cherry-pick that first commit into a separate PR so that we could merge this first and have a clean baseline for further fixes? This would reduce the risk of conflicts when backporting fixes into 4.3.x.

**Comment by JankKeks** [comment]
> > You renamed the binding already, but the internal path was not renamed - that caused a lot of internal warnings and problems with the mapping and classes. So I fixed it also.
> 
> Thanks for clarifying and fixing this. Would you be able to cherry-pick that first commit into a separate PR so that we could merge this first and have a clean baseline for further fixes? This would reduce the risk of conflicts when backporting fixes into 4.3.x.

Unfortunately I have no idea how to do that :) If your provide some hints, I could do that.
Regarding back porting - I already build it for 4.3.0 - and it runs fine.

**Comment by jlaur** [comment]
> Unfortunately I have no idea how to do that :) If your provide some hints, I could do that.

Sure, here is how I would do it:
```shell
git checkout main
git pull
git checkout -b freeathome-fix-path
git cherry-pick a4be1ebe726da112378cf2703ad208f1d7d799fd
git push --set-upstream origin freeathome-fix-path
# now create PR from this branch
# when merged, update your fork, then:
git checkout main
git pull
git checkout THISBRANCH
git rebase main
```

And now I see that you committed your changes directly to your main branch, so you will not be able to do any of this without getting into trouble. Making changes directly on the main branch is a bad idea, because you will not be able to work on different branches at the same time, and the probability of running into issues with rebasing etc. is high.

It is possible to resolve this, but it requires some more advanced git commands where any mistake can lead to bad results. If you are not already comfortable with using git this way, I would rather not try to give advise, because I could make mistakes as well and not be able to support you if it goes sideways.

If @lsiepel is okay with backporting the full contents of this PR, we could keep it as one PR and merge it as is (assuming PR comments are resolved). Going forward after that, you should create PR's from separate branches rather than using main.

@andrasU - would you like to review this PR as binding maintainer?

**Comment by JankKeks** [comment]
> > Unfortunately I have no idea how to do that :) If your provide some hints, I could do that.
> 
> Sure, here is how I would do it:
> 
> ```shell
> git checkout main
> git pull
> git checkout -b freeathome-fix-path
> git cherry-pick a4be1ebe726da112378cf2703ad208f1d7d799fd
> git push --set-upstream origin freeathome-fix-path
> # now create PR from this branch
> # when merged, update your fork, then:
> git checkout main
> git pull
> git checkout THISBRANCH
> git rebase main
> ```
> 
> And now I see that you committed your changes directly to your main branch, so you will not be able to do any of this without getting into trouble. Making changes directly on the main branch is a bad idea, because you will not be able to work on different branches at the same time, and the probability of running into issues with rebasing etc. is high.
> 
> It is possible to resolve this, but it requires some more advanced git commands where any mistake can lead to bad results. If you are not already comfortable with using git this way, I would rather not try to give advise, because I could make mistakes as well and not be able to support you if it goes sideways.
> 
> If @lsiepel is okay with backporting the full contents of this PR, we could keep it as one PR and merge it as is (assuming PR comments are resolved). Going forward after that, you should create PR's from separate branches rather than using main.
> 
> @andrasU - would you like to review this PR as binding maintainer?

Ok thank you, I will keep it in mind for the next change. Unfortunately I'd never worked with git before o.o

**Review by lsiepel — APPROVED** [review]
Thanks, LGTM. 
Will backport all of it, that is fine.

**Pull request merged**